lib/active_shipping/shipment_event.rb

```diff
@@ -13,5 +13,10 @@ def delivered?
     def status
       @status ||= name.downcase.gsub("\s", "_").to_sym
     end
+
+    def ==(other)
+      name == other.name && time == other.time && location == other.location && message == other.message &&
+        type_code == other.type_code
```

Comment:
Given the number of things you're comparing, I'd prefer it if we separated them by lines, i.e.

``` ruby
name == other.name &&
time == other.time &&
location == other.location &&
...
```

Alternatively, you could iterate over all of them, like:

``` ruby
attributes = [:name, :time, :location, :message, :type_code]
attributes.all? { |attr| attr == other.public_send(:attr) }
```